src/System.Threading.Thread/src/System/Threading/Thread.cs

```diff
@@ -190,14 +193,31 @@ public static IPrincipal CurrentPrincipal
         {
             get
             {
-                return CurrentThread._principal ?? (CurrentThread._principal = AppDomain.CurrentDomain.GetThreadPrincipal());
+                if (t_principal is null)
+                {
+                   CurrentPrincipal = AppDomain.CurrentDomain.GetThreadPrincipal();
```

Comment:
Nit: looks like the indentation here is slightly off

Reply:
fixed